sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/impl/DiscountServiceImpl.java

```diff
@@ -62,4 +67,19 @@ public DiscountDto update(DiscountDto discount) {
     public void delete(Long id) {
         discountDAO.deleteById(id);
     }
+
+    @Override
+    public Page<DiscountDto> getByCriteria(DiscountSearchCriteria searchCriteria) {
+        String searchText = "%" + searchCriteria.getSearchText() + "%";
+        List<Discount> result;
+        if (searchCriteria.getTags() == null) {
+            result = discountDAO.getByCriteria(searchText, searchCriteria.getRate());
+        } else {
+            result = discountDAO.getByCriteriaWithTags(searchText,
+                    searchCriteria.getTags(), searchCriteria.getRate());
+        }
+        List<DiscountDto> discountDtos = mapper.mapAll(result, DiscountDto.class);
+        return new PageImpl<>(discountDtos, searchCriteria.getPageRequest(), discountDtos.size());
```

Comment:
Also if it is required we should put tags data loaded by set of discount ids to result DTO list